### DISCUSSION TITLE
[Idea] Auto-select tax-rules when creating products

### DISCUSSION THREAD

**Discussion by LarsGMortensen** [PrestaShop/PrestaShop#33863] [Ideas]
### The problem

In my country (Denmark) everything sold to anybody is always (without exception) taxed with 25% VAT (called "moms" in Denmark). That is always and without exception.

So, if you run a shop in Denmark (and possibly other countries?), it seems quite redundant that you manually must select the Tax-rule each and everytime you create a product in PrestaShop:

![tax-rule-select](https://github.com/PrestaShop/PrestaShop/assets/26773745/2a712dd5-0e8b-4e52-9a76-ace9c91b66ff)

And furthermore, if you forget to select it, and a customer buys the product, you will per defintion loose 20% of the products value (as you will have to deduct the tax), which can potentially be quite a lot of money.

So, it would be SO NICE! if it was somehow possible to make PrestaShop pre-select the tax-rule, so that it would just automatically be added whenever you create a product.

### The solution

Make a feature that enables the shop-owner to configure PrestaShop to always add the tax-rule when creating a product.

Maybe something like this:

![tax-add-by-default](https://github.com/PrestaShop/PrestaShop/assets/26773745/53f1aa76-02c2-4d21-9cb5-60c54a34bdde)

![add-by-default-2](https://github.com/PrestaShop/PrestaShop/assets/26773745/654f439e-cea1-4330-aafd-609bc43eb62d)

And basically this should just pre-select that tax-rule when creating a new product.

### Alternatives

I have tried to fix it myself, as I planned to add the "selected"-attribute to the correct tax-rule:

![image](https://github.com/PrestaShop/PrestaShop/assets/26773745/33d4850d-1ee9-4f0e-8a00-b0404178bb13)

However, I cant find the particular select-dropdown anywhere in the codebase of PrestaShop (using Notepad++ I have searched for files including "product_pricing_retail_price_tax_rules_group_id" but there were no results).

But if anybody could let me know how I could add this myself (i.e. tell me where that particular select-dropdown is generated), I would be happy to just fix it myself.

### Do you plan to work on this subject?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [X] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by MatShir** [comment]
Great idea, it could set on the product setting ! The best would be to make product template

**Reply by LarsGMortensen** [reply]
Hi @MatShir 

"it could set on the product setting" 

Can you elaborate? :)

"The best would be to make product template"

Can you elaborate? :)

**Reply by MatShir** [reply]
- In product setting or on tax page, there could be a configuration where you set the default tax for each new product created. 
- The idea of template will go further (bigger project I guess), to pre save product type, features values, tax, categories and more into a template. So when you create a product, you could use a presaved template directly. Just an idea

**Reply by LarsGMortensen** [reply]
Right, now I get what you mean by "template": Defining a set of "default" configurations to be set when creating a new product (i.e. defining that the tax should always be applied). That could probably be awesome and make the everyday life with PrestaShop much better for store-owners!

But other than setting the tax I am not quite sure what I would use it for - because almost every other setting really varies from product to product for me. Some products are simple products, some are with combinations, some are in this category some are in that category, etc, etc. So mostly I just really miss being able to auto-set the tax code :)

So, to me it would really just be super awesome if I just had a toggle under "tax-rules", something like this:

![tax-add-by-default](https://github.com/PrestaShop/PrestaShop/assets/26773745/25f01118-46fc-474b-afcb-cde743093553)

![add-by-default-2](https://github.com/PrestaShop/PrestaShop/assets/26773745/4ba52ee7-1e76-4c1e-abc1-137586b1b2a3)

**Reply by kpodemski** [reply]
@MatShir this behavior differs from the old product page and it was reported to me as well, on the old product page tax is already selected

**Reply by MatShir** [reply]
The old page the first tax rule was pre-selected, if i remember well

**Comment by MatShir** [comment]
Fixed by https://github.com/PrestaShop/PrestaShop/pull/35495 in coming soon (in few days) next path release 8.1.5. To follow the release, check the [issue](https://github.com/PrestaShop/PrestaShop/issues/35216).
